phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -354,7 +354,12 @@ private List<PTable> getClientSideMaintainedIndexes(TableRef tableRef) {
         }
         return Collections.emptyList();
     }
-    
+
+    /**
+     * Implementation of MutationPlan that is selected if
+     * 1) the query either has no WHERE clause or it is strictly point lookup, and
```

Comment:
> Just `point lookup` is correct in my opinion.

now i understand why this is true, even though there is no client indexes, for this to be true, there is at least one query plan available
`noQueryReqd &= queryPlans.size() == 1 + clientSideIndexes.size();`

and when noQueryReqd is true, it will at least go through this once:
```
            noQueryReqd = (!context.getScan().hasFilter()
                    || context.getScan().getFilter() instanceof SkipScanFilter)
                && context.getScanRanges().isPointLookup();
```

hence, i agree that point lookup is necessary for MultiRowDeleteMutation to get selected.